src/mame/drivers/uknc.cpp

```diff
@@ -9,97 +9,946 @@ Graphics 640x288 pixels.
 
 2009-05-12 Skeleton driver.
 
-Status: both CPUs start in the weeds.
+Status: snapshot of work in progress.  needs real VM2 CPU core.
 
 ****************************************************************************/
 
 #include "emu.h"
+
+#include "bus/qbus/qbus.h"
 #include "cpu/t11/t11.h"
+#include "imagedev/cassette.h"
+#include "machine/bankdev.h"
+#include "machine/timer.h"
+
 #include "emupal.h"
 #include "screen.h"
 
 
+#define UKNC_TOTAL_HORZ (800)
+#define UKNC_DISP_HORZ  (640)
+#define UKNC_HORZ_START (80)
+
+#define UKNC_TOTAL_VERT (312)
+#define UKNC_DISP_VERT  (288+18)
+#define UKNC_VERT_START (0)
+
+
+typedef struct
+{
+	uint16_t addr;
+	bool cursor;
+	uint32_t aux;
+	bool color;
+	bool control;
+	bool cursor_type;
+	int cursor_color;
+	int cursor_octet;
+	int cursor_pixel;
+} scanline_t;
```

Comment:
Names ending in `_t` are reserved at global scope by POSIX.  Also, be careful what you put at global scope – you can always nest things or use anonymous namespaces.

Reply:
renamed the type.